crates/pathfinder/src/state/merkle_tree.rs

```diff
@@ -1570,6 +1651,566 @@ mod tests {
+                    values,
+                    root,
+                    tree,
+                }
+            }
+
+            /// Calls `get_proof` and `verify_proof` on every key/value pair in the random_tree.
+            fn verify(&self) {
+                let keys_bits: Vec<&BitSlice<Msb0, u8>> =
+                    self.keys.iter().map(|k| k.view_bits()).collect();
+                let proofs = get_proofs(&keys_bits, &self.tree).unwrap();
+                keys_bits
+                    .iter()
+                    .zip(self.values.iter())
+                    .enumerate()
+                    .for_each(|(i, (k, v))| {
+                        let verified = verify_proof(self.root, k, *v, &proofs[i]).unwrap();
+                        assert_eq!(verified, Membership::Member, "Failed to prove key");
+                    });
+            }
+        }
+
+        /// Generates a storage proof for each `key` in `keys` and returns the result in the form of an array.
+        fn get_proofs<'a, 'tx>(
+            keys: &'a [&BitSlice<Msb0, u8>],
+            tree: &MerkleTree<RcNodeStorage<'tx, '_>>,
+        ) -> anyhow::Result<Vec<Vec<ProofNode>>> {
+            keys.iter().map(|k| tree.get_proof(k)).collect()
+        }
+
+        #[test]
+        fn simple_binary() {
+            let mut conn = rusqlite::Connection::open_in_memory().unwrap();
+            let transaction = conn.transaction().unwrap();
+            let mut uut = MerkleTree::load("test", &transaction, StarkHash::ZERO).unwrap();
+
+            //   (250, 0, x1)
+            //        |
+            //     (0,0,x1)
+            //      /    \
+            //     (2)  (3)
+
+            let key_1 = starkhash!("00"); // 0b01
+            let key_2 = starkhash!("01"); // 0b01
+
+            let key1 = key_1.view_bits();
+            let key2 = key_2.view_bits();
+            let keys = [key1, key2];
+
+            let value_1 = starkhash!("02");
+            let value_2 = starkhash!("03");
+
+            uut.set(key1, value_1).unwrap();
+            uut.set(key2, value_2).unwrap();
+            let root = uut.commit().unwrap();
+
+            let uut = MerkleTree::load("test", &transaction, root).unwrap();
+
+            let proofs = get_proofs(&keys, &uut).unwrap();
+
+            let verified_key1 = verify_proof(root, key1, value_1, &proofs[0]).unwrap();
+
+            assert_eq!(verified_key1, Membership::Member);
+        }
+
+        #[test]
+        fn double_binary() {
+            let mut conn = rusqlite::Connection::open_in_memory().unwrap();
+            let transaction = conn.transaction().unwrap();
+            let mut uut = MerkleTree::load("test", &transaction, StarkHash::ZERO).unwrap();
+
+            //           (249,0,x3)
+            //               |
+            //           (0, 0, x3)
+            //         /            \
+            //     (0,0,x1)       (1, 1, 5)
+            //      /    \          /     \
+            //     (2)  (3)       (0)    (5)
```

Comment:
The `(1,1,5)` node is actually an edge node, so I think the `(0)` leaf should be removed from the diagram.

```suggestion
            //     (0,0,x1)       (1, 1, 5)
            //      /    \             |
            //     (2)  (3)           (5)
```